controllers/restic.go

```diff
@@ -53,16 +56,23 @@ func (r *VeleroReconciler) ReconcileResticDaemonset(log logr.Logger) (bool, erro
 	ds := &appsv1.DaemonSet{
 		ObjectMeta: getResticObjectMeta(r),
 	}
-
 	if velero.Spec.EnableRestic != nil && !*velero.Spec.EnableRestic {
-
-		if err := r.Get(r.Context, r.NamespacedName, ds); err == nil {
+		deleteContext := context.Background()
+		if err := r.Get(deleteContext, types.NamespacedName{
+			Name:      ds.Name,
+			Namespace: r.NamespacedName.Namespace,
+		}, ds); err == nil {
 			// no errors means there already is an existing DaeMonset.
 			// TODO: Check if restic is in use, a backup is running, so don't blindly delete restic.
 			// If velero Spec enableRestic exists and is false, attempt to delete.
-			if err := r.Delete(r.Context, ds); err != nil {
+			deleteOptionPropagationForeground := metav1.DeletePropagationForeground
+			if err := r.Delete(deleteContext, ds, &client.DeleteOptions{PropagationPolicy: &deleteOptionPropagationForeground}); err != nil {
+				r.EventRecorder.Event(ds, v1.EventTypeNormal, "DeleteDaemonSetFailed", "Got DaemonSet to delete but could not delete err:"+err.Error())
 				return false, err
 			}
+			r.EventRecorder.Event(ds, v1.EventTypeNormal, "DeletedDaemonSet", "DaemonSet deleted")
```

Comment:
I think that we should have more consistent names for this with higher level buckets